code/Weapons/Gadget/Components/ExplosiveGadgetComponent.cs

```diff
@@ -99,7 +104,8 @@ public virtual void Explode()
 		}
 
 		ExplodeSoundClient( To.Everyone, ExplosionSound );
-		Gadget.Delete();
+		if ( DeleteOnExplode || !Gadget.Tags.Has( Tag.Invincible ) )
```

Comment:
I think we can ignore the `!Gadget.Tags.Has( Tag.Invincible)` here, the prefab bool is good.

Reply:
Agreed.